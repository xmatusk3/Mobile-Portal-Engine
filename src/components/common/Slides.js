import React, { Component } from 'react';
import { View, Text, ScrollView, Dimensions, TouchableWithoutFeedback } from 'react-native';
import { Button } from 'react-native-elements';

const SCREEN_WIDTH = Dimensions.get('window').width;

class Slides extends Component {
  _renderLastSlide = index => {
    if (index === this.props.data.length - 1) {
      return (
        <Button
          title="Onwards!"
          containerViewStyle={styles.buttonContainerStyle}
          buttonStyle={styles.buttonStyle}
          onPress={this.props.onComplete}
          raised
        />
      );
    }
    return null;
  };
  _renderSlides = () => {
    return this.props.data.map((text, index) => {
      return (
        <View key={index} style={styles.slideStyle}>
          <Text style={styles.textStyle}>{text}</Text>
          {this._renderLastSlide(index)}
        </View>
      );
    });
  };

  render() {
    return (
      <ScrollView horizontal pagingEnabled>
        {this._renderSlides()}
      </ScrollView>
    );
  }
}

const styles = {
  buttonStyle: {
    backgroundColor: '#497d04',
    shadowColor: '#355e00',
    shadowOffset: { height: -3, width: 0 },
  },
  buttonContainerStyle: {
    backgroundColor: '#497d04',
    marginTop: 15,
  },
  slideStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: SCREEN_WIDTH,
    backgroundColor: '#ededed',
  },
  textStyle: {
    fontSize: 30,
    textAlign: 'center',
    color: '#2a2a2a',
  },
};

export default Slides;
