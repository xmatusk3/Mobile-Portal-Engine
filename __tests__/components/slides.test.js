import React from 'react';
import Slides from '../../src/components/common/Slides.js';

import renderer from 'react-test-renderer';

describe('Slides component', () => {
  describe('renders', () => {
    it('correctly one slide', () => {
      const slideText = ['TestText'];
      const slides = renderer.create(<Slides data={slideText} onComplete={jest.fn()} />).toJSON();

      expect(slides).toMatchSnapshot();
    });

    it('correctly multiple slides', () => {
      const slidesText = ['TestText1', 'TestText2'];
      const slides = renderer.create(<Slides data={slidesText} onComplete={jest.fn()} />).toJSON();

      expect(slides).toMatchSnapshot();
    });
  });
});
