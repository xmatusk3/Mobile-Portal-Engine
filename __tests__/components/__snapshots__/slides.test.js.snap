// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slides component renders correctly multiple slides 1`] = `
<RCTScrollView
  horizontal={true}
  pagingEnabled={true}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "flex": 1,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2a2a2a",
            "fontSize": 30,
            "textAlign": "center",
          }
        }
      >
        TestText1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "flex": 1,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2a2a2a",
            "fontSize": 30,
            "textAlign": "center",
          }
        }
      >
        TestText2
      </Text>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "backgroundColor": "#497d04",
              "marginTop": 15,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#497d04",
                  "shadowColor": "#355e00",
                  "shadowOffset": Object {
                    "height": -3,
                    "width": 0,
                  },
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Onwards!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Slides component renders correctly one slide 1`] = `
<RCTScrollView
  horizontal={true}
  pagingEnabled={true}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ededed",
          "flex": 1,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2a2a2a",
            "fontSize": 30,
            "textAlign": "center",
          }
        }
      >
        TestText
      </Text>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "backgroundColor": "#497d04",
              "marginTop": 15,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#497d04",
                  "shadowColor": "#355e00",
                  "shadowOffset": Object {
                    "height": -3,
                    "width": 0,
                  },
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Onwards!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
